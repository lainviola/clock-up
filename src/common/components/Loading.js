// @flow
import React from 'react'
import styled from 'styled-components'
import { BounceLoader } from 'react-spinners'

type Props = {}

export class Loading extends React.Component<Props> {
  render() {
    return (
      <Container>
        <BounceLoader color={'#36D7B7'} size={100} />
      </Container>
    )
  }
}

const Container = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`

export default Loading
