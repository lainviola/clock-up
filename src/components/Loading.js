// @flow
import React from 'react'

type Props = {}

export class Loading extends React.Component<Props> {
  render() {
    return <div>Loading</div>
  }
}

export default Loading
