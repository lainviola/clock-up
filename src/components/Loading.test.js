import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Loading } from './Loading'

describe('<Loading />', () => {
  const setup = function() {
    const wrapper = shallow(<Loading />)
    return wrapper
  }

  it('should be render', () => {
    const wrapper = setup()
    expect(wrapper.exists()).toBe(true)
  })

  it('should be match snapshot', () => {
    const wrapper = setup()

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
