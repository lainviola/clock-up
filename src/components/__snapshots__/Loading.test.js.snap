// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should be match snapshot 1`] = `
<Loading__Container>
  <onlyUpdateForKeys(Loader)
    color="#36D7B7"
    loading={true}
    size={100}
  />
</Loading__Container>
`;
